src/drone/agent/executor.rs

```diff
@@ -89,7 +90,8 @@ impl Executor {
         let _span = tracing::span!(
             Level::INFO,
             "run_backend",
-            backend_id = spawn_request.backend_id.id()
+            backend_id = spawn_request.backend_id.id(),
+            metadata = %json!(spawn_request.metadata),
```

Comment:
We need to pass this to the `Executing state.` log below, too.

Reply:
Executing state runs while we have entered this span, so it is automatically added.

Reply:
According to [these docs](https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code) it sounds like this span might be open while other async tasks are running, causing that metadata to incorrectly get applied to other log messages. Am I understanding that correctly?